### PR TITLE
feat: migrate Toaster component

### PR DESCRIPTION
**Pull request by Raubzeug**
add Toaster component
add Readme in English

### CODE REVIEW

File: src/components/Toaster/Toast/Toast.tsx [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+                }
+            };
+
+            return (
+                <Link key={`${label}__${index}`} className={b('action')} onClick={onActionClick}>
+                    {label}
+                </Link>
+            );
+        });
+    };
```
Comment (ValeraS) [inline src/components/Toaster/Toast/Toast.tsx:222]: let's extract this function and rename

```typescript
function renderActions({actions, onClose})
```

File: src/components/Toaster/Toast/Toast.tsx [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    const getTitleIcon = () => {
+        const {type} = props;
+        const icon = type ? TITLE_ICONS[type] : null;
+
+        if (!icon) {
+            return null;
+        }
+
+        return <Icon data={icon} className={b('icon', {title: true})} />;
+    };
```
Comment (ValeraS) [inline src/components/Toaster/Toast/Toast.tsx:182]: let's extract this function:

```typescript
function renderIcon({type})
```

**Review by ValeraS — CHANGES_REQUESTED** [review]

File: src/components/Toaster/Toast/Toast.tsx [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+interface UseCloseOnTimeoutProps {
+    onClose: VoidFunction;
+    timeout?: number;
+}
+
+function useCloseOnTimeout({onClose, timeout}: UseCloseOnTimeoutProps): {
+    onMouseLeave: VoidFunction;
+    onMouseOver: VoidFunction;
+} {
```
Comment (ValeraS) [inline src/components/Toaster/Toast/Toast.tsx:72]: don't have to specify return type, typescript can infers it automatically
```suggestion
function useCloseOnTimeout({onClose, timeout}: UseCloseOnTimeoutProps) {
```

File: src/components/Toaster/Toast/Toast.tsx [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+        containerProps: {onAnimationEnd},
+        handleClose,
+    } = useToastStatus({onRemove: props.removeCallback});
+
+    const height = useHeight({ref, isOverride});
+
+    const containerProps = useCloseOnTimeout({
+        onClose: handleClose,
+        timeout: allowAutoHiding ? props.timeout || DEFAULT_TIMEOUT : undefined,
+    });
```
Comment (ValeraS) [inline src/components/Toaster/Toast/Toast.tsx:233]:
```suggestion
    const timeout = allowAutoHiding ? props.timeout || DEFAULT_TIMEOUT : undefined
    const closeOnTimeoutProps = useCloseOnTimeout({onClose: handleClose, timeout});
```

File: src/components/Toaster/Toast/Toast.tsx [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+
+    const getStyles = () => {
+        const styles: ToastStyles = {};
+
+        if (height && status !== ToastStatus.showingIndents && status !== ToastStatus.shown) {
+            styles.height = height;
+        }
+
+        if (status !== 'creating') {
+            styles.position = 'relative';
```
Comment (ValeraS) [inline src/components/Toaster/Toast/Toast.tsx:243]: maybe set it by css class, for example, add mod `creating`

File: src/components/Toaster/Toast/Toast.tsx [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+        onClose: handleClose,
+        timeout: allowAutoHiding ? props.timeout || DEFAULT_TIMEOUT : undefined,
+    });
+
+    const getStyles = () => {
+        const styles: ToastStyles = {};
+
+        if (height && status !== ToastStatus.showingIndents && status !== ToastStatus.shown) {
+            styles.height = height;
+        }
```
Comment (ValeraS) [inline src/components/Toaster/Toast/Toast.tsx:240]: let's move it to useHeight, and change:

```typescript
function useToastHeight({isOveride}): {style, ref}

const heightProps = useToastHeight({isOveride});
return <div {...heightProps}>
```

File: src/components/Toaster/Toast/Toast.tsx [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+}
+
+export function Toast(props: ToastProps) {
+    const ref = React.useRef<HTMLDivElement>(null);
+
+    const {allowAutoHiding = true, isClosable = true, isOverride = false} = props;
+
+    const {
+        status,
+        containerProps: {onAnimationEnd},
```
Comment (ValeraS) [inline src/components/Toaster/Toast/Toast.tsx:224]:
```suggestion
        containerProps: statusProps,
```

File: src/components/Toaster/Toast/Toast.tsx [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+
+    const {content, actions, title, className, type} = props;
+    return (
+        <div
+            ref={ref}
+            className={b(mods, className)}
+            style={getStyles()}
+            onAnimationEnd={onAnimationEnd}
+            {...containerProps}
+        >
```
Comment (ValeraS) [inline src/components/Toaster/Toast/Toast.tsx:280]:
```suggestion
        <div
            className={b(mods, className)}
            {...statusProps}
            {...heightProps}
            {...closeOnTimeoutProps}
        >
```

File: src/components/Toaster/Toast/Toast.tsx [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+            <div className={b('title', {bold: Boolean(content || actions)})}>
+                {renderIcon({type})}
+                {title}
+            </div>
+            {getCloseButton()}
```
Comment (ValeraS) [inline src/components/Toaster/Toast/Toast.tsx:285]:
```suggestion
            {isClosable && (
             <Button
                view="flat-secondary"
                size="s"
                style={{position: 'absolute', top: 10, right: 10}}
                onClick={handleClose}
            >
                <Icon data={CrossIcon} />
            </Button>
       )}
```

**Review by ValeraS — CHANGES_REQUESTED** [review]

File: src/components/Toaster/Toast/Toast.tsx [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+
+    React.useEffect(() => {
+        setHeight(getToastHeight());
+    }, [getToastHeight]);
+
+    React.useEffect(() => {
+        if (isOverride) {
+            setHeight(getToastHeight());
+        }
+    }, [isOverride, getToastHeight]);
```
Comment (ValeraS) [inline src/components/Toaster/Toast/Toast.tsx:129]: This doesn't  work if override twice with different content. 

```suggestion
    React.useEffect(() => {
        if (isOverride) {
            setHeight(getToastHeight());
        }
    });
```

**Review by ValeraS — CHANGES_REQUESTED** [review]

File: src/components/Toaster/Toast/Toast.tsx [hunk 11]
```diff
@@ -0,0 +1,263 @@
...
+}
+
+function useToastHeight({isOverride, status}: UseToastHeightProps) {
+    const [height, setHeight] = React.useState<number | undefined>(undefined);
+
+    const ref = React.useRef<HTMLDivElement>(null);
+
+    const getToastHeight = React.useCallback(() => {
+        return ref.current?.offsetHeight;
+    }, [ref]);
```
Comment (ValeraS) [inline src/components/Toaster/Toast/Toast.tsx:114]:
```suggestion
    }, []);
```

**Review by ValeraS — APPROVED** [review]
one more small fix and I think it's ready to go

File: src/components/Toaster/Toast/Toast.tsx [hunk 12]
```diff
@@ -0,0 +1,263 @@
...
+}
+
+interface ToastProps extends ToastGeneralProps, ToastInnerProps {}
+
+enum ToastStatus {
+    creating = 'creating',
+    showingIndents = 'showing-indents',
+    showingHeight = 'showing-height',
+    hiding = 'hiding',
+    shown = 'shown',
```
Comment (amje) [inline src/components/Toaster/Toast/Toast.tsx:56]: Enum name starts with capital letter

File: src/components/Toaster/__stories__/ToasterShowcase.tsx [hunk 13]
```diff
@@ -0,0 +1,337 @@
...
+    createSuccessToast = () => {
+        const {showCloseIcon, setTimeout, timeout, allowAutoHiding, setContent, setActions} =
+            this.state;
+
+        const name = this.getToastName('success');
+
+        this.toaster.createToast({
+            name,
+            type: 'success',
+            title: 'Оплата успешно произведена',
```
Comment (amje) [inline src/components/Toaster/__stories__/ToasterShowcase.tsx:96]: Let's rewrite all file texts in english

**Review by amje — CHANGES_REQUESTED** [review]

**Review by amje — APPROVED** [review]

**Pull request merged**